File: experimental/ingraph-rete/src/main/java/hu/bme/mit/ingraph/rete/nodes/AbstractJoinNode.java
package hu.bme.mit.ingraph.rete.nodes;

import org.eclipse.collections.api.multimap.MutableMultimap;
import org.eclipse.collections.impl.factory.Multimaps;

import hu.bme.mit.ingraph.rete.data.Tuple;

public abstract class AbstractJoinNode {

	protected MutableMultimap<Tuple<?>, Tuple<?>> leftIndexer = Multimaps.mutable.set.empty();
	protected MutableMultimap<Tuple<?>, Tuple<?>> rightIndexer = Multimaps.mutable.set.empty();
	
}

File: ingraph-engine/ingraph-engine-ire/src/main/scala/hu/bme/mit/ire/nodes/binary/AbstractJoinNode.scala
package hu.bme.mit.ire.nodes.binary

import hu.bme.mit.ire.datatypes._

abstract class AbstractJoinNode(val primaryMask: Mask,
                                val secondaryMask: Mask) extends BinaryNode {

  def extract(tuple: Tuple, mask: Mask): Tuple = {
    mask.map(i => tuple(i))
  }

}
